rnd/autogpt_server/autogpt_server/data/analytics.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+import prisma.enums
+import prisma.types
+
+logger = logging.getLogger(__name__)
+
+
+async def log_raw_analytics(
+    user_id: str,
+    type: prisma.enums.AnalyticsType,
+    data: dict,
+    data_index: str,
+):
+    details = await prisma.models.AnalyticsDetails.prisma().create(
+        data={
+            "userId": user_id,
+            "type": type,
+            "data": prisma.Json(data),
+            "dataIndex": data_index,
+        }
+    )
+    return details
+
+
+async def log_raw_metric(
+    user_id: str,
+    metric_name: prisma.enums.AnalyticsMetric,
+    aggregation_type: prisma.enums.AggregationType,
+    metric_value: float,
+    data_string: str,
+):
+    if metric_value < 0:
+        raise ValueError("metric_value must be non-negative")
+
+    if aggregation_type == prisma.enums.AggregationType.NO_AGGREGATION:
+        value_increment = metric_value
+        counter_increment = 0
+    elif aggregation_type in [
+        prisma.enums.AggregationType.COUNT,
+        prisma.enums.AggregationType.SUM,
+    ]:
+        value_increment = metric_value
+        counter_increment = 1
+    elif aggregation_type in [
+        prisma.enums.AggregationType.AVG,
+        prisma.enums.AggregationType.MAX,
+        prisma.enums.AggregationType.MIN,
+    ]:
+        value_increment = 0  # These will be handled differently in a separate query
+        counter_increment = 1
+    else:
+        raise ValueError(f"Unsupported aggregation_type: {aggregation_type}")
+
+    result = await prisma.models.AnalyticsMetrics.prisma().upsert(
+        data={
+            "update": {
+                "value": {"increment": value_increment},
```

Comment:
Do we need this line then?

Reply:
if we find an exact match of name, id, data, and agg we increment both of these and move on. in some cases we don't increment value here, and instead handle it in the below block, but one or the other is alwasy incremented